File: Python/Semana9/Leccion07/capa_datos_persona/Persona.py
class Persona:
    #Método Dunder Init
    def __init__(self, id_persona, nombre, apellido, email):
        self._id_persona = id_persona
        self._nombre = nombre
        self._apellido = apellido
        self._email = email

    #Método Dunder Str
    def __str__(self):
        return f'''
            Id Persona: {self._id_persona},
            Nombre: {self._nombre},Apellido: {self._apellido},
            Email: {self._email}
        '''
    
    #Métodos Getters and Setters
    @property
    def id_persona(self):
        return self._id_persona
    
    @id_persona.setter
    def id_persona(self, id_persona):
        self._id_persona = id_persona

    @property
    def nombre(self):
        return self._nombre
    
    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre

    @property
    def apellido(self):
        return self._apellido
    
    @apellido.setter
    def apellido(self, apellido):
        self._apellido = apellido
    
    @property
    def email(self):
        return self._email
    
    @email.setter
    def email(self, email):
        self._email = email
       

File: JavaScript/Semana10/TSLeccion07/Persona.js
class Persona{

    static contadorPersonas = 0; //atributo estático

    constructor(nombre, apellido, edad){ //propiedades de instancia.
        this._idPersona = ++Persona.contadorPersonas;
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }

    //Métodos Getters and setters

    //solo el getter para el atributo idPersona
    get idPersona(){
        return this._idPersona;
    }

    //getter y setter para el atributo nombre
    get nombre(){
        return this._nombre;
    }

    set nombre(nombre){
        this._nombre = nombre;
    }

    //getter y setter para el atributo apellido
    get apellido(){
        return this._apellido;
    }

    set apellido(apellido){
        this._apellido = apellido;
    }

    //getter y setter para el atributo edad
    get edad(){
        return this._edad;
    }

    set edad(edad){
        this._edad = edad;
    }

    //agregamos el toString()
    toString(){ 
        return ` //Dato: al convertir a template String, asi mismo se ve en la consola
        ${this._idPersona} 
        ${this._nombre} 
        ${this._apellido} 
        ${this._edad}`;
    }

}